### PR TITLE
[Feature] Improve chat freeze

### PR DESCRIPTION
**Pull request by DayBr3ak**
Currently, if the chatBuffer is full, messages will drop even if the chat is "frozen". 

This PR delays messages for ever, as long as the chat is frozen. Messages are queued back when autoscrolling resumes.

### CODE REVIEW

**Comment by DayBr3ak** [comment]
doesn't work on subs / gift

**Comment by DayBr3ak** [comment]
The new version extend the buffer size.

File: src/modules/chat_freeze/index.js [hunk 1]
```diff
@@ -22,6 +56,7 @@ function shouldFreeze(e) {
 
 class ChatFreezeModule {
     constructor() {
+        let keysPressed = 0;
```
Comment (night) [inline src/modules/chat_freeze/index.js]: why is this moved

File: src/modules/chat_freeze/index.js [hunk 2]
```diff
@@ -1,12 +1,46 @@
...
+}
+
+function patchedSetState({isAutoScrolling}) {
+    if (isAutoScrolling !== undefined) {
+        setChatBufferSize(isAutoScrolling ? bufferSize : Infinity);
+    }
+    return twitchSetState.apply(this, arguments);
+}
+
+function patchScroller() {
```
Comment (night) [inline src/modules/chat_freeze/index.js]: we should be able to do this without monkeypatching, so i'd rather do it without monkeypatching.

Comment (DayBr3ak) [reply src/modules/chat_freeze/index.js]: If you don't monkey patch, and "click" on the `More messages below`, you're stuck with you're infite value, running into memory issues.

Comment (DayBr3ak) [reply src/modules/chat_freeze/index.js]: I guess I could listen to the click on that element.

Comment (DayBr3ak) [reply src/modules/chat_freeze/index.js]: No sometimes you can resume scrolling without clicking on the element. Monkeypatching seems to be the 'safe' way.
Should I drop this PR? @night

Comment (night) [reply src/modules/chat_freeze/index.js]: i dont think you need to drop it.. just needs to be fixed to be safer.. you can detect when scrolling is happening with js events .. not sure why we need to patch to know if user is scrolling or not

**Comment by night** [comment]
closing due to inactivity
